backend/h5p/H5PContentStorage.ts

```diff
@@ -0,0 +1,453 @@
+     * to be created with addContent(...) first.
+     * @param contentId The id of the content to add the file to
+     * @param filename The filename; can be a path including subdirectories
+     * (e.g. 'images/xyz.png')
+     * @param readStream A readable stream that contains the data
+     * @param user (optional) The user who owns this object
+     */
+    async addFile(
+        contentId: ContentId,
+        filename: string,
+        readStream: Readable,
+        user?: IUser
+    ): Promise<void> {
+        if (!(await this.canUserAccessContent(user.id, contentId))) {
+            throw new LoernwerkError(
+                'User isnt allowed to edit this content',
+                LoernwerkErrorCodes.FORBIDDEN
+            );
+        }
+
+        // Converting the readable to a string
+        const chunks = [];
+        for await (const chunk of readStream) {
+            chunks.push(chunk);
+        }
+        const fileBuffer = Buffer.concat(chunks);
+
+        const dbFile = new DBH5PFile();
+        dbFile.filename = filename;
+        dbFile.ownerType = 'content';
+        dbFile.owner = contentId;
+        dbFile.content = fileBuffer;
+        dbFile.size = fileBuffer.length;
+        await dbFile.save();
+    }
+
+    /**
+     * Checks if a piece of content exists in storage.
+     * @param contentId the content id to check
+     * @returns true if the piece of content exists
+     */
+    async contentExists(contentId: ContentId): Promise<boolean> {
+        const content = await DBH5PContent.findOneBy({
+            h5pContentId: contentId,
+        });
+        return content !== null;
+    }
+
+    /**
+     * Deletes a content object and all its dependent files from the repository.
+     * Throws errors if something goes wrong.
+     * @param contentId The content id to delete.
+     * @param user The user who wants to delete the content
+     */
+    async deleteContent(contentId: ContentId, user?: IUser): Promise<void> {
+        if (!(await this.canUserAccessContent(user.id, contentId, true))) {
+            throw new LoernwerkError(
+                'User isnt allowed to edit this content',
+                LoernwerkErrorCodes.FORBIDDEN
+            );
+        }
+
+        await DBH5PContent.delete({ h5pContentId: contentId });
+        await DBH5PFile.delete({ ownerType: 'content', owner: contentId });
+    }
+
+    /**
+     * Deletes a file from a content object.
+     * @param contentId the content object the file is attached to
+     * @param filename the file to delete; can be a path including
+     * subdirectories (e.g. 'images/xyz.png')
+     * @param user The user who wants to delete the file
+     */
+    async deleteFile(
+        contentId: ContentId,
+        filename: string,
+        user?: IUser
+    ): Promise<void> {
+        if (!(await this.canUserAccessContent(user.id, contentId, true))) {
+            throw new LoernwerkError(
+                'User isnt allowed to edit this content',
```

Comment:
`'` missing in message

Reply:
I have been in a prettier-eslint war about this - If I add the ', I've got to escape it, since we're using single quotes outside aswell, but prettier then uses double quotes outside to allow the ' inside without escaping it, but eslint then sees this as an issue, because we want single quotes

Therefore, I'm unsure how to fix this without prettier-ignore or something

Reply:
does \' work?

Reply:
or just write `is not`